utils/mysql.go

```diff
@@ -139,3 +141,121 @@ func MySQLInstallReplication(mysqlDir string, installPath string, port int) {
+	dbUser := options["dbUser"]
+	rwUser := options["rwUser"]
+	roUser := options["roUser"]
+	remoteAccess := options["remoteAccess"]
+	dbPassword := options["dbPassword"]
+	replUser := options["replUser"]
+	replPassword := options["replPassword"]
+	grantsMySQLFormat := `use mysql;
+set password=password(%s);
+grant all on *.* to %s@'%s' identified by '%s';
+grant all on *.* to %s@'localhost' identified by '%s';
+grant SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,INDEX,ALTER,
+    SHOW DATABASES,CREATE TEMPORARY TABLES,LOCK TABLES, EXECUTE 
+    on *.* to %s@'localhost' identified by '%s';
+grant SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,INDEX,ALTER,
+    SHOW DATABASES,CREATE TEMPORARY TABLES,LOCK TABLES, EXECUTE 
+    on *.* to %s@'%s' identified by '%s';
+grant SELECT,EXECUTE on *.* to %s@'%s' identified by '%s';
+grant SELECT,EXECUTE on *.* to %s@'localhost' identified by '%s';
+grant REPLICATION SLAVE on *.* to %s@'%s' identified by '%s';
+delete from user where password='';
+delete from db where user='';
+flush privileges;
+create database if not exists test;
+`
+	grantsMysql := fmt.Sprintf(grantsMySQLFormat, dbPassword, dbUser, remoteAccess, dbPassword, dbUser, dbPassword, rwUser, dbPassword, rwUser, remoteAccess, dbPassword, roUser, remoteAccess, dbPassword, roUser, dbPassword, replUser, remoteAccess, replPassword)
+	scripts["grants.mysql"] = grantsMysql
+	grants576MySQLFormat := `use mysql;
+set password='%s';
+-- delete from tables_priv;
+-- delete from columns_priv;
+-- delete from db;
+delete from user where user not in ('root', 'mysql.sys', 'mysqlxsys');
+
+flush privileges;
+
+create user %s@'%s' identified by '%s';
+grant all on *.* to %s@'%s' ;
+
+create user %s@'localhost' identified by '%s';
+grant all on *.* to %s@'localhost';
+
+create user %s@'localhost' identified by '%s';
+grant SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,INDEX,ALTER,
+     SHOW DATABASES,CREATE TEMPORARY TABLES,LOCK TABLES, EXECUTE 
+     on *.* to %s@'localhost';
+
+create user %s@'%s' identified by '%s';
+grant SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,INDEX,ALTER,
+    SHOW DATABASES,CREATE TEMPORARY TABLES,LOCK TABLES, EXECUTE 
+    on *.* to %s@'%s';
+     
+create user %s@'%s' identified by '%s';
+create user %s@'localhost' identified by '%s';
+create user %s@'%s' identified by '%s';
+grant SELECT,EXECUTE on *.* to %s@'%s';
+grant SELECT,EXECUTE on *.* to %s@'localhost';
+grant REPLICATION SLAVE on *.* to %s@'%s';
+create schema if not exists test;
+`
+	grants576Mysql := fmt.Sprintf(grants576MySQLFormat, dbPassword, dbUser, remoteAccess, dbPassword, dbUser, remoteAccess, dbUser, dbPassword, dbUser, rwUser, dbPassword, rwUser, rwUser, remoteAccess, dbPassword, rwUser, remoteAccess, roUser, remoteAccess, dbPassword, roUser, dbPassword, replUser, remoteAccess, replPassword, roUser, remoteAccess, roUser, replUser, remoteAccess)
+	scripts["grants_5_7_6.mysql"] = grants576Mysql
+}
+
+func MySQLGrantsMySQL(installPath string, code string, code576 string) {
```

Comment:
MySQLGrantsMySQL function just install one grant script, not for master and slave together,
and you can define another function MySQLInstallRepGrantFile which can install two grant scripts
for both master and slave.

MySQLGrantsMySQL need not generate two grant files for two mysql version, cause you can 
use GetMySQLVersion to determine which grant script is really needed.

you should make every function is simple and just do one thing.